components/profile-controller/controllers/profile_controller.go

```diff
@@ -104,12 +108,15 @@ type ProfileReconciler struct {
 // Automatically generate RBAC rules to allow the Controller to read and write Deployments
 func (r *ProfileReconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
 	logger := r.Log.WithValues("profile", request.NamespacedName)
-	defaultKubeflowNamespaceLabels := r.readDefaultLabelsFromFile(r.DefaultNamespaceLabelsPath)
-
+	defaultKubeflowNamespaceLabels, err := r.readDefaultLabelsFromFiles(r.DefaultNamespaceLabelsPaths)
+	if err != nil {
+		r.ShutdownFileWatchers()
+		os.Exit(1)
```

Comment:
I kept the exit if loading the labels file fails, but it might be worth trying to come up with a new solution later since this could cause issues if we later configure multiple concurrent reconciles.